bdk-ffi/src/wallet.rs

```diff
@@ -522,4 +540,71 @@ mod test {
         assert_eq!(tx_details.fee.unwrap(), 110);
         assert!(tx_details.confirmation_time.is_none());
     }
+
+    #[test]
+    fn test_get_address() {
+        let test_wpkh = "wpkh(tprv8hwWMmPE4BVNxGdVt3HhEERZhondQvodUY7Ajyseyhudr4WabJqWKWLr4Wi2r26CDaNCQhhxEftEaNzz7dPGhWuKFU4VULesmhEfZYyBXdE/0/*)";
+        let descriptor = Descriptor::new(test_wpkh.to_string(), Network::Regtest).unwrap();
+        let change_descriptor = Descriptor::new(
+            test_wpkh.to_string().replace("/0/*", "/1/*"),
+            Network::Regtest,
+        )
+        .unwrap();
+
+        let wallet = Wallet::new(
+            Arc::new(descriptor),
+            Some(Arc::new(change_descriptor)),
+            Network::Regtest,
+            DatabaseConfig::Memory,
+        )
+        .unwrap();
+
+        assert_eq!(
+            wallet
+                .get_address(crate::AddressIndex::New)
+                .unwrap()
+                .address,
+            "bcrt1qqjn9gky9mkrm3c28e5e87t5akd3twg6xezp0tv"
+        );
+
+        assert_eq!(
+            wallet
+                .get_address(crate::AddressIndex::New)
+                .unwrap()
+                .address,
+            "bcrt1q0xs7dau8af22rspp4klya4f7lhggcnqfun2y3a"
+        );
+
+        assert_eq!(
+            wallet
+                .get_address(crate::AddressIndex::LastUnused)
+                .unwrap()
+                .address,
+            "bcrt1q0xs7dau8af22rspp4klya4f7lhggcnqfun2y3a"
+        );
+
+        assert_eq!(
+            wallet
+                .get_internal_address(crate::AddressIndex::New)
+                .unwrap()
+                .address,
+            "bcrt1qpmz73cyx00r4a5dea469j40ax6d6kqyd67nnpj"
+        );
+
+        assert_eq!(
+            wallet
+                .get_internal_address(crate::AddressIndex::New)
+                .unwrap()
+                .address,
+            "bcrt1qaux734vuhykww9632v8cmdnk7z2mw5lsf74v6k"
+        );
+
+        assert_eq!(
+            wallet
+                .get_internal_address(crate::AddressIndex::LastUnused)
+                .unwrap()
+                .address,
+            "bcrt1qpmz73cyx00r4a5dea469j40ax6d6kqyd67nnpj"
```

Comment:
The tests are failing because the address here should be `bcrt1qaux734vuhykww9632v8cmdnk7z2mw5lsf74v6k`. Makes sense since it's not being used in between the two assertions, and so `LastUnused` should return the same address as the previous call on 596

Reply:
You're absolutely right. Sorry, I'll try to check the tests more closely going forwards.